### PR TITLE
app/vmselect: fix panic when using `/select/multitenant` endpoint

### PR DESCRIPTION
**Pull request by zekker6**
Such requests must be rejected as not found since vmselect does not support multitenant endpoint.

See: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4910

### CODE REVIEW

**Review by Amper — APPROVED** [review]
LGTM

File: app/vmselect/main.go [hunk 1]
```diff
@@ -292,6 +292,11 @@ func requestHandler(w http.ResponseWriter, r *http.Request) bool {
 		httpserver.Errorf(w, r, "auth error: %s", err)
 		return true
 	}
+	if at == nil {
+		// the only option for at to be nil is when p.AuthToken == "multitenant"
```
Comment (f41gh7) [inline app/vmselect/main.go:296]: I think, explicit error must be returned in this case. That vmselect doesn't support multitenant endpoint.

**Review by hagen1778 — APPROVED** [review]

File: app/vmselect/main.go (hunk 1 above)
Comment (hagen1778) [reply app/vmselect/main.go:296]: Before this change it printed:
```
remoteAddr: "127.0.0.1:64861"; requestURI: /select/multitenant/prometheus; unsupported path requested: "/select/multitenant/prometheus"
```
Now it prints:
```
remoteAddr: "127.0.0.1:64515"; requestURI: /select/multitenant/prometheus; auth error: tenantID must be specified explicitly in the request path
```


I find the first better. Wdyt @f41gh7 ?

**Review by f41gh7 — APPROVED** [review]
LGTM

**Pull request merged**